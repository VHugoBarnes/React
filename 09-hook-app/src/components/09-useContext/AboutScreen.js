import React from 'react';

export const AboutScreen = () => {
    return (
        <div>
            <h1 className="mt-3 ml-3">AboutScreen</h1>
            <hr/>
        </div>
    );
};
