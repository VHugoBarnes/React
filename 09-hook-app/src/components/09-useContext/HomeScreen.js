import React from 'react';

export const HomeScreen = () => {
    return (
        <div>
            <h1 className="mt-3 ml-3">HomeScreen</h1>
            <hr/>
        </div>
    );
};
