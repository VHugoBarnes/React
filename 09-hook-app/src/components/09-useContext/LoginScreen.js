import React from 'react';

export const LoginScreen = () => {
    return (
        <div>
            <h1 className="mt-3 ml-3">LoginScreen</h1>
            <hr/>
        </div>
    );
};
