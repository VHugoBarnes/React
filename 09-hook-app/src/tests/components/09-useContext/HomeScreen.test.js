import React from 'react';
import '@testing-library/jest-dom';
import { mount, shallow } from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en <HomeScren />', () => {
   
    const user = {
        name: 'VÃ­ctor',
        email: 'hugovg799@gmail.com',
    };

    const wrapper = mount(
        <UserContext.Provider value={{user}}>
            <HomeScreen/>
        </UserContext.Provider>
    );

    test('Debe mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });
    
});
