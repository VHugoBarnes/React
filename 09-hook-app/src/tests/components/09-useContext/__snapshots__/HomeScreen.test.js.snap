// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScren /> Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
    "name": "VÃ­ctor",
    "email": "hugovg799@gmail.com"
}
  </pre>
</div>
`;
